summingbird-batch/src/main/scala/com/twitter/summingbird/batch/BatchID.scala

```diff
@@ -93,7 +93,7 @@ object BatchID {
       .flatMap {
         case (min, max, cnt) =>
           if ((min + cnt) == (max + 1L)) {
-            Some(Interval.leftClosedRightOpen(min, max.next).right.get)
+            Some(Intersection(InclusiveLower(min), InclusiveUpper(max)))
```

Comment:
Looks like some other tests started to fail with this fix, which mean we have code which *assumes* interval is left closed right open. Fix you proposed seems reasonable for me (especially for real world).